### PR TITLE
Refactor backend

### PR DESCRIPTION
**Pull request by Wenzel**
This is a an experiment to developer a better API than our so called "`Backend`".

I wanted to abstract the `get_backend` method, and hide it, and i wanted to use the `Nitro` name for the main nitro API.

So i renamed  `nitro.py` to `listener.py`
I created a new class `Nitro` which contains the vcpu `listener` and the `backend`.

Therefore, we can now have this API
~~~Python
nitro = Nitro(domain)
nitro.listener.set_traps(True)
for event in nitro.listen()
     nitro.backend.process_event(event)
~~~

Introspection is enabled by default, but you can disable it.
~~~Python
nitro = Nitro(domain, False)
~~~

This is just a draft for a new API.

What do you think @Soft ? :)

### CODE REVIEW

File: nitro/nitro.py [hunk 1]
```diff
@@ -1,155 +1,21 @@
...
-        'stop_request',
-        'futures',
-        'queue',
-        'current_cont_event',
-    )
+    def __init__(self, domain, introspection=True):
+        self.listener = Listener(domain)
+        self.introspection = introspection
+        if self.introspection:
+            self.backend = get_backend(domain)
```
Comment (Soft) [inline nitro/nitro.py]: I think it would be nicer to set backend to None instead of leaving it out completelly if introspection is not enabled. This would also enable us to leave out the introspection property and just test if backend is None in stop. Of course this is all pretty subjective.

**Comment by Soft** [comment]
I like these changes, I think they make the API cleaner. I think I'll merge them in a bit.

File: nitro/nitro.py (hunk 1 above)
Comment (Wenzel) [reply nitro/nitro.py]: Agreed.
Commit has been pushed.
